From: Simon McVittie <smcv@debian.org>
Date: Mon, 4 Nov 2019 17:29:16 +0000
Subject: Make ld executable configurable

Tools like this should be configurable in a cross or native file. In
particular, if we are cross-compiling (with an executable wrapper like
qemu-arm), the build system ld is not necessarily able to manipulate
host system objects.

Signed-off-by: Simon McVittie <smcv@collabora.com>
Applied-upstream: 2.63.2, commit:81936ca5808ed7f52571e1fdc49ecffc4d202b07
---
 docs/reference/glib/cross.xml | 1 +
 gio/tests/meson.build         | 5 +++--
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/docs/reference/glib/cross.xml b/docs/reference/glib/cross.xml
index 977421f..c90b30b 100644
--- a/docs/reference/glib/cross.xml
+++ b/docs/reference/glib/cross.xml
@@ -59,6 +59,7 @@ c_link_args = []
 c = 'x86_64-w64-mingw32-gcc'
 cpp = 'x86_64-w64-mingw32-g++'
 ar = 'x86_64-w64-mingw32-ar'
+ld = 'x86_64-w64-mingw32-ld'
 objcopy = 'x86_64-w64-mingw32-objcopy'
 strip = 'x86_64-w64-mingw32-strip'
 pkgconfig = 'x86_64-w64-mingw32-pkg-config'
diff --git a/gio/tests/meson.build b/gio/tests/meson.build
index 2c84a20..8b5d105 100644
--- a/gio/tests/meson.build
+++ b/gio/tests/meson.build
@@ -586,8 +586,9 @@ if not meson.is_cross_build() or meson.has_exe_wrapper()
   # (https://reviews.llvm.org/D58234). FIXME: This test could be enabled for
   # LLVM once that support is in a stable release.
   objcopy = find_program('objcopy', required : false)
+  ld = find_program('ld', required : false)
 
-  if build_machine.system() == 'linux' and cc.get_id() == 'gcc' and objcopy.found()
+  if build_machine.system() == 'linux' and cc.get_id() == 'gcc' and objcopy.found() and ld.found()
     test_gresource_binary = custom_target('test5.gresource',
       input : 'test5.gresource.xml',
       output : 'test5.gresource',
@@ -616,7 +617,7 @@ if not meson.is_cross_build() or meson.has_exe_wrapper()
     test_resources_binary = custom_target('test_resources.o',
       input : test_gresource_binary,
       output : 'test_resources.o',
-      command : ['ld',
+      command : [ld,
                  '-r',
                  '-b','binary',
                  '@INPUT@',
